valentyusb/usbcore/test/common.py

```diff
@@ -20,8 +21,33 @@ def grouper(n, iterable, pad=None):
     """
     return zip_longest(*[iter(iterable)]*n, fillvalue=pad)
 
-
-class CommonUsbTestCase(unittest.TestCase):
+class BaseUsbTestCase(unittest.TestCase):
+    """
+    Test case helpers common to all test cases, simple and complex
+    """
+    def make_vcd_name(self, basename=None, modulename=None):
```

Comment:
Rather then supply basename, could we instead supply an "extra" component that way we don't have to duplicate the basename values in these other tests?

Reply:
Good idea.  I've added a `testsuffix=` optional parameter, which gets appended to the (usually auto-derived) basename.  And replaced all the `basename=...` overrides with that.  I've left `basename` as a parameter override in case we need it later (overrides `self.id()`) since it's a tiny bit of code.  But currently only the `modulename=` override is used directly.

This updates the names of a number of the vcd output files again (to include class names / method names too), but the predictability of naming is probably useful anyway.

Ewen